### PR TITLE
New post: "When are two methods alike?"

### PR DESCRIPTION
**Pull request by S11001001**
Second in a series on type parameters and members, for posting on July 16.

NB: Links to the previous post will not work on GH, because they use `post_url`, which only works on the site.

### CODE REVIEW

File: _posts/2015-07-16-method-equiv.md [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+type parameter—or a method type parameter lowered to an
+existential—but there are other pairs of method types I want to
+explore that are the same, or very close.  So let’s talk about how we
+determine this equivalence.
+
+A method *r* is more general than or as general as *q* if *q* may be
+implemented by only making a call to *r*, passing along the arguments.
+By more general, we mean *r* can be invoked in all the situations that
+*q* can be invoked in, and more besides.  Let us call the result of
+this test *r* <:*ₘ* *q*, where <:*ₘ* is pronounced “party duck”; if
```
Comment (non) [inline _posts/2015-07-16-method-equiv.md]: Maybe make the "where <:*ₘ* is pronounced 'party duck'" part parenthetical?

> Let us call the result of this test *r* <:*ₘ* *q* (where <:*ₘ* is pronounced “party duck”).

File: _posts/2015-07-16-method-equiv.md [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+the test of *q* making a call to  *r* fails, then ¬(*r* <:*ₘ* *q*).
+
+If *q* <:*ₘ* *r* and *r* <:*ₘ* *q*, then the two method types are
+*equivalent*; that is, neither has more expressive power than the
+other, since each can be implemented merely by invoking the other and
+doing nothing else.  We write this as *q* ≡*ₘ* *r*.  Likewise, if
+*r* <:*ₘ* *q* and ¬(*q* <:*ₘ* *r*), that is, *q* can be written by
+calling *r*, but not vice versa, then *r* is *strictly more general*
+than *q*, or *r* <:*ₘ* *q*.
```
Comment (non) [inline _posts/2015-07-16-method-equiv.md]: The way I read this, you are saying that there are two relations, *party duck* and *strictly-party duck*. But I see the same symbol (<:*ₘ*) being used for both. Should one be <≡:*ₘ*?

**Comment by tpolecat** [comment]
Whatever character you have after the `<:` is just showing up as a rectangle for me. Is it supposed to be something else?

**Comment by non** [comment]
@tpolecat I see a triple equals (congruence) symbol.

**Comment by S11001001** [comment]
@tpolecat It's an italicized, subscripted "m", to draw an analogy with the "weak conformance" symbol from SLS, which uses a little "w" instead.  Can you tell me more about the font it's using?

File: _posts/2015-07-16-method-equiv.md (hunk 1 above)
Comment (S11001001) [inline _posts/2015-07-16-method-equiv.md]: @non Thanks, adopted that form.

File: _posts/2015-07-16-method-equiv.md (hunk 2 above)
Comment (S11001001) [inline _posts/2015-07-16-method-equiv.md]: @non No, but the example of the latter was wrong.  Fixed, thanks.

**Comment by tpolecat** [comment]
Yeah it shows up as `LATIN SUBSCRIPT SMALL LETTER M` in the special chars dialog if i paste it in, but it doesn't appear to be present in any font that I have. It's using Helvetica Neue.

**Comment by S11001001** [comment]
@tpolecat Including the one on typelevel.org posts?  It only really matters there, I suppose.  I'm loath to give up the symbol, but alternate encodings would be nice.

**Comment by tpolecat** [comment]
Yeah it's the same font there. Maybe we could turn on the TeX formatting doodad that they use for SLS. I don't know how that works.

**Comment by S11001001** [comment]
@tpolecat That might be pretty, but I'll try getting away with &lt;sub> and &lt;small> first, e.g. *<sub><small>m</small></sub>*.

(Sure, they work on GH, but jekyll is what matters.)

**Comment by larsrh** [comment]
You can enable LaTeX markup by setting `mathjax: true` in the front matter of the page (just like `pygments: true`). There should be some examples in the existing posts. Also, it's in the README :-)

**Comment by S11001001** [comment]
@larsrh Hah, I didn't notice that, thank you!

I'll experiment and share a screenshot here; see if you like how it is in mathjax better than it is now (@tpolecat it should look good to you now).

**Comment by S11001001** [comment]
Q and R capitalized but left italic when standing alone, expressions translated to mathjax, DejaVu fonts.

http://i.imgur.com/sHnprNq.jpg

Compared to those paragraphs in the current draft, how does it look to you?

**Comment by tpolecat** [comment]
Yep, looks ok now. I think the mathjax version is prettier and easier to read, although I would probably go ahead and encode the standalone Q/R.

**Comment by S11001001** [comment]
OK, total mathjax has landed.

**Comment by S11001001** [comment]
As usual, though, I've just α-converted until it works, and maybe these variable names in the definitions are less clear.

**Comment by S11001001** [comment]
@tpolecat Do the new subscripts in 6d5ab21 look OK to you?

**Comment by tpolecat** [comment]
@S11001001 looks great.

**Comment by non** [comment]
:+1:

**Comment by S11001001** [comment]
Published.  Thank you @non, @tpolecat, @larsrh for reviewing!

**Pull request merged**